src/floating.c

```
@@ -647,7 +647,7 @@ void floating_resize_window(Con *con, const bool proportional,
         floating_reposition(con, initial_rect);
 
     /* If this is a scratchpad window, don't auto center it from now on. */
-    if (con->scratchpad_state == SCRATCHPAD_FRESH)
+    if (con_exists(con) && con->scratchpad_state == SCRATCHPAD_FRESH)
```

Are the checks here and below actually necessary? And if they are, I think it'd better to do a »fail fast« approach:

```
if (!con_exists(con)) {
    DLOG("The container has been closed in the meantime.\n");
    return;
}
```

This would also help reintroducing the bug by referencing `con` somewhere else without remembering to add the check.

---

The checks are necessary since this is the code that gets executed right after the floating disable command.

> This would also help reintroducing the bug by referencing con somewhere else without remembering to add the check.

How else could this be handled?